### PR TITLE
Introduce new states

### PR DESCRIPTION
**Pull request by lynxis**
based on branch small misc. Please merge #42 first.

### CODE REVIEW

File: client/asyncns.c [hunk 1]
```diff
@@ -77,7 +77,7 @@ enum {
 };
 
 struct asyncns {
-    int fds[MESSAGE_FD_MAX];
+    int fds[4];
```
Comment (mitar) [inline client/asyncns.c]: Please use verbose-named constants instead of just numbers.

**Comment by mitar** [comment]
#42 has been merged.

File: client/asyncns.c (hunk 1 above)
Comment (lynxis) [inline client/asyncns.c]: this is part of a syncronisation of the upstream project of this file.

Comment (mitar) [inline client/asyncns.c]: Oh, this is 3rd party code? Why it is not in some `vendor` subdirectory or something. :-)

Comment (mitar) [inline client/asyncns.c]: Have you submitted `#define _GNU_SOURCE` upstream as well?

Comment (lynxis) [inline client/asyncns.c]: No, because they are using autoconf to define their config.h and a lot of defines.

Comment (mitar) [inline client/asyncns.c]: Hm, but if we update this code next time from upstream, how can we know what to reapply? Maybe we should have some patch and apply it during our compilation? Or we should try to get those things upstream.

**Comment by lynxis** [comment]
ToDos before merging:
[] fix comments (/* */ by //)
[] drop state ERROR

**Comment by lynxis** [comment]
ready for merge.

**Comment by mitar** [comment]
Could you comment somewhere (maybe the best in the commit message) which revision of upstream `asyncns.c ` has this now been? Because then later on we can easier make a diff against it, see what are our changes and apply them.

Even better it is if we have commit with original revision on its own in the history. Then when you want to update the upstream, you go to that commit again, you update the version, commit new (you are in detached head mode then), and then merge back into the latest of our repository. In this way then git then handles all our merging of our changes on top.

**Comment by lynxis** [comment]
ping @kostko @mitar

File: client/l2tp_client.c [hunk 2]
```diff
@@ -1097,6 +1151,8 @@ void context_process(l2tp_context *ctx)
       }
       break;
     }
+    case STATE_IDLE:
```
Comment (kostko) [inline client/l2tp_client.c]: Please add braces as it is in other `case` statements.

**Comment by kostko** [comment]
I see some fixes in the external asyncns. Are we testing DNS resolution?

**Comment by kostko** [comment]
Apart from the minor comment above, this looks good to me. It would be very nice if we could also have tests for DNS resolution, but this is something that can be added in another PR.

**Comment by mitar** [comment]
I think we are testing DNS resolution in 8bbc5e4a226f73aa990ff9a08c0441ac38bea4b8?

**Pull request merged**